extensions/kafka-connect/src/main/java/com/hazelcast/jet/kafka/connect/impl/KafkaConnectSource.java

```diff
@@ -18,47 +18,51 @@
 
 import com.hazelcast.core.HazelcastException;
 import com.hazelcast.jet.pipeline.SourceBuilder;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
 import org.apache.kafka.connect.connector.ConnectorContext;
 import org.apache.kafka.connect.source.SourceConnector;
 import org.apache.kafka.connect.source.SourceRecord;
-import org.apache.kafka.connect.source.SourceTask;
-import org.apache.kafka.connect.source.SourceTaskContext;
-import org.apache.kafka.connect.storage.OffsetStorageReader;
 
-import java.util.HashMap;
+import javax.annotation.Nonnull;
 import java.util.List;
 import java.util.Map;
 import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
 
 import static com.hazelcast.client.impl.protocol.util.PropertiesUtil.toMap;
 import static com.hazelcast.internal.util.Preconditions.checkRequiredProperty;
 import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
 
 public class KafkaConnectSource {
+    private static final ILogger LOGGER = Logger.getLogger(KafkaConnectSource.class);
 
     private final SourceConnector connector;
-    private final SourceTask task;
-    private final Map<String, String> taskConfig;
+
+    private final TaskRunner taskRunner;
+
     /**
      * Key represents the partition which the record originated from. Value
      * represents the offset within that partition. Kafka Connect represents
      * the partition and offset as arbitrary values so that is why it is
      * stored as map.
      * See {@link SourceRecord} for more information regarding the format.
      */
-    private Map<Map<String, ?>, Map<String, ?>> partitionsToOffset = new HashMap<>();
-    private boolean taskInitialized;
+    private Map<Map<String, ?>, Map<String, ?>> partitionsToOffset = new ConcurrentHashMap<>();
```

Comment:
Maybe make the field final ?

Reply:
Fixed https://github.com/hazelcast/hazelcast/pull/23695/commits/750e2b5981a2ccc13653d818b7f559626b845f9f